pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -361,6 +361,13 @@ public Future<Void> sendMessages(final List<? extends Entry> entries, EntryBatch
                 msgOutCounter.add(totalMessages);
                 bytesOutCounter.add(totalBytes);
                 chunkedMessageRate.recordMultipleEvents(totalChunkedMessages, 0);
+            } else {
+                log.warn("[{}-{}] Sent messages to client fail by IO exception[{}], these messages(messages count:"
```

Comment:
Have you considered the connection probelm will print tons of logs here?

Reply:
> Have you considered the connection problems that will print tons of logs here? :)
> IMO, the message trace logs should be considered carefully.

Added a health check here, if the health check fail, this connection will be closed immediately. So this log only be printed when the health check is in progress.